deploy-sg-lambda.yml

```diff
@@ -0,0 +1,102 @@
+  vars:
+    ansible_connection: local
+    lambda_name: cf-security-group-update
+    lambda_function_name: cf-security-group-update-test
+    lambda_handler: "{{lambda_name}}.lambda_handler"
+    security_group_name: cloudflare_lambda_demo_sg
+    output_dir: ~/ansible_testing
+
+  tasks:
+    - name: set up AWS credentials
+      block:
+        - set_fact:
+          aws_connection_info: &aws_connection_info
+            aws_region: '{{ aws_region }}'
+            aws_access_key: '{{ aws_access_key }}'
+            aws_secret_key: '{{ aws_secret_key }}'
+            security_token: '{{ security_token }}'
+          no_log: true
+
+    - tags:
+        - given that we have a lambda installed
+      block:
+
+        - name: ensure that output dir exists
+          file:
+            path: "{{output_dir}}"
+            state: directory
+            mode: 0775
+
+        # ============================================================
+        # direct zip file upload
+        - name: move lambda into place for archive module
+          copy:
+            src: "{{lambda_name}}.py"
+            dest: "{{output_dir}}/{{lambda_name}}.py"
+
+        - name: bundle lambda into a zip
+          archive:
+            format: zip
+            path: "{{output_dir}}/{{lambda_name}}.py"
+            dest: "{{output_dir}}/{{lambda_name}}.zip"
+          register: zip_res
+
+        # This should exist, but there's no expectation that the test user should be able to
+        # create/update this role, merely validate that it's there.
+        # Use ansible -m iam_role -a 'name=ansible_lambda_role
+        # assume_role_policy_document={{ lookup("file",
+        # "test/integration/targets/lambda_policy/files/minimal_trust_policy.json", \
+        #        convert_data=False) }}
+        # ' -vvv localhost
+        # to create this through more privileged credentials before running this test suite.
+        # - name: create minimal lambda role
+        #   iam_role:
+        #     name: ansible_lambda_role
+        #     assume_role_policy_document: "{{ lookup('file', 'minimal_trust_policy.json', convert_data=False) }}"
+        #     create_instance_profile: no
+        #     <<: *aws_connection_info
+        #   register: iam_role
+
+        # - name: wait 10 seconds for role to become available
+        #   pause:
+        #     seconds: 10
+        #   when: iam_role.changed
+
+        - name: ensure security group is present
+          ec2_group:
+            name: "{{ security_group_name }}"
+            state: present
+            description: security group for testing cloudflare ip lambda
+            <<: *aws_connection_info
+          register: group_create
+
+        - name: test state=present - upload the lambda
+          lambda:
+            name: "{{lambda_function_name}}"
+            runtime: "python3.6"
+            handler: "{{lambda_handler}}"
+            role: "arn:aws:iam::455225740156:role/service-role/basicLambdaRole"
```

Comment:
Hardcoded account ID? Be nice to have as a param, even though just for testing

Reply:
It's overridable / overridden in deploy, but I will make this look nicer and add IAM role setup to the repo somehow.